lib/rules/no-misleading-character-class.js

```diff
@@ -208,21 +309,23 @@ module.exports = {
                 return;
             }
 
-            const foundKinds = new Set();
+            const foundKindMatches = [];
 
             visitRegExpAST(patternNode, {
                 onCharacterClassEnter(ccNode) {
                     for (const chars of iterateCharacterSequence(ccNode.elements)) {
                         for (const kind of kinds) {
-                            if (hasCharacterSequence[kind](chars)) {
-                                foundKinds.add(kind);
+                            const matches = accumulate(chars, characterSequenceIndexFilters[kind]);
+
+                            if (matches.length) {
+                                foundKindMatches.push({ kind, matches });
```

Comment:
I think that `foundKindMatches` should be a `Map` where the key is `kind` and the value is `matches`. Then here we could make sure that all matches of the same kind are appended to the same `matches` array.

```suggestion
                                if (foundKindMatches.has(kind)) {
                                    foundKindMatches.get(kind).push(...matches);
                                } else {
                                    foundKindMatches.set(kind, matches);
                                }
```

And the iteration over `kind` and `matches` would become `for (const [kind, matches] of foundKindMatches)`...

This would fix the duplicate problem report for `new RegExp("[ \\ufe0f][ \\ufe0f]")`.